### PR TITLE
chore: add release automation

### PR DESCRIPTION
**Pull request by seebees**
Added verdaccio publish testing.
Added CodeBuild spec to version and publish releases.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by texastony — APPROVED** [review]
My only comment is that `verdaccio/config.yaml` looks scary, since it references an empty password and has comments that indicate that anyone can publish our packages...

But I think that is just my ignorance. I am certain that we use verdaccio as a mocked npm registry, and that is what those settings are for.

Later, we use verdaccio again to actually publish to npm... as long as I am right about that, :shipit:

**Comment by seebees** [comment]
`verdaccio` is used to verify the publishing process. So before we publish to real npm we use `verdaccio` to test publish and see that everything will work.

File: .github/workflows/ci-unit-tests.yaml [hunk 1]
```diff
@@ -13,7 +13,6 @@ jobs:
           - windows-latest
           - macos-latest
         node:
-          - 8
```
Comment (farleyb-amazon) [inline .github/workflows/ci-unit-tests.yaml:16]: Not removing 10 as well?

**Review by farleyb-amazon — APPROVED** [review]

File: .github/workflows/ci-unit-tests.yaml (hunk 1 above)
Comment (seebees) [reply .github/workflows/ci-unit-tests.yaml:16]: I'll remove that when we get the policy into this repo

**Pull request merged**